src/blueapi/plans/plans.py

```diff
@@ -93,5 +97,16 @@ def count(
     Yields:
         Iterator[MsgGenerator]: _description_
     """
+    plan_args = {
+        "detectors": list(map(repr, detectors)),
+        "num": num,
+    }
+    if delay:  # If bp.count added delay to plan_args, we could remove all md handling
+        plan_args.update({"delay": delay})
```

Comment:
For optional plan args I'm in favour of allowing `"arg": null` in the JSON rather than maybe not having it there at all. It's generally easier to deserialize if the documents are all consistent for a given plan.